tools/clang/include/clang/SPIRV/Type.h

```diff
@@ -51,57 +57,66 @@ class Type {
   bool isCompositeType() const;
   bool isImageType() const;
 
-  static const Type *getType(SPIRVContext &ctx, spv::Op op,
-                             std::vector<uint32_t> arg = {},
-                             std::set<const Decoration *> decs = {});
-
-  static const Type *getVoid(SPIRVContext &ctx);
-  static const Type *getBool(SPIRVContext &ctx);
-  static const Type *getInt8(SPIRVContext &ctx);
-  static const Type *getUint8(SPIRVContext &ctx);
-  static const Type *getInt16(SPIRVContext &ctx);
-  static const Type *getUint16(SPIRVContext &ctx);
-  static const Type *getInt32(SPIRVContext &ctx);
-  static const Type *getUint32(SPIRVContext &ctx);
-  static const Type *getInt64(SPIRVContext &ctx);
-  static const Type *getUint64(SPIRVContext &ctx);
-  static const Type *getFloat16(SPIRVContext &ctx);
-  static const Type *getFloat32(SPIRVContext &ctx);
-  static const Type *getFloat64(SPIRVContext &ctx);
+  static const Type *getVoid(SPIRVContext &ctx, DecorationSet decs = {});
```

Comment:
But should remove the `DecorationSet` parameters on primitive types. They are not able to be decorated in my understanding.